libr/cons/cons.c

```diff
@@ -521,14 +521,15 @@ R_API void r_cons_filter() {
 
 R_API void r_cons_push() {
 	if (I.cons_stack) {
-		if (I.buffer_len < 1) {
-			return;
-		}
 		RConsStack *data = R_NEW0 (RConsStack);
-		data->buf = malloc (I.buffer_len);
-		if (!data->buf) {
-			free (data);
-			return;
+		if (I.buffer_len > 0) {
+			data->buf = malloc (I.buffer_len);
+			if (!data->buf) {
+				free (data);
+				return;
+			}
+		} else {
+			data->buf = NULL;
```

Comment:
Practically, `I.buffer_len == 0` when `data->buf == NULL` so it should not segfault, but I can't find that behavior stated anywhere, and http://en.cppreference.com/w/c/string/byte/memcpy states that the behavior is undefined, so I've moved `memcpy` inside the `if` clause.